.github/workflows/ci.yml

```
@@ -83,3 +86,47 @@ jobs:
 
       - name: Run unit tests
         run: pnpm -r test
+
+  ensraninbow-docker-build:
+    runs-on: ubuntu-latest
+    needs: [static-analysis, unit-tests]
+    if: github.event_name == 'push' && (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/alpha' || github.ref == 'refs/heads/subgraph')
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Build and push Docker image
+        env:
+          IMAGE_NAME: ghcr.io/${{ github.repository }}/ensrainbow
+          TAG: ${{ github.ref == 'refs/heads/main' && 'latest' || github.ref == 'refs/heads/alpha' && 'alpha' || github.ref == 'refs/heads/subgraph' && 'subgraph' }}
+        run: |
+          echo ${{ secrets.GITHUB_TOKEN }} | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+          docker build -f apps/ensrainbow/Dockerfile -t $IMAGE_NAME:$TAG -t $IMAGE_NAME:${{ github.sha }} .
+          docker save $IMAGE_NAME:$TAG $IMAGE_NAME:${{ github.sha }} > image.tar
+
+      - name: Upload image artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: docker-image
+          path: image.tar
+
+  ensrainbow-docker-push:
+    runs-on: ubuntu-latest
+    needs: [ensraninbow-docker-build]
+    
+    steps:
+      - name: Download image artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: docker-image
+          
+      - name: Load and push Docker image
```

```suggestion
      - name: Load and push Docker image to GitHub
```

Is that right? Goal is to make it more explicit where it is being pushed to

It's also not so clear to me what's the difference between "Load and push Docker image" and "Build and push Docker image" -- but I'm really weak at DevOps. Just sharing my thoughts.